certification/internal/shell/scorecard_basic-check-spec.go

```diff
@@ -0,0 +1,174 @@
+    
+    err = os.MkdirAll(bundleDir, 0777)
+    if err != nil {
+        logger.Error("unable to create subdirs: ", err)
+        return false, err
+    }
+
+    stdouterr, err := exec.Command("oc", "image", "extract", "--path", "/:"+bundleDir, bundleImage).CombinedOutput()
+    if err != nil {
+        logger.Error("unable to execute oc image extract on the image: ", err)
+        return false, err
+    }
+
+    // Need to be in the parent directory of the bundle contents
+    err = os.Chdir(tmpDir)
+    if err != nil {
+        logger.Error("unable to change dir: ", err)
+        return false, err
+    }
+    
+    configFile := filepath.Join(bundleDir,"/tests/scorecard/config.yaml")
+
+    //copy config file to the root
+    input, err := ioutil.ReadFile(configFile)
+    if err != nil {
+        logger.Error("unable to read config.yaml: ", err)
+        return false, err
+    }
+    
+    destinationFile := filepath.Join(tmpDir,"/config.yaml")
+    
+    err = ioutil.WriteFile(destinationFile, input, 0777)
+    if err != nil {
+        logger.Error("unable to write to config.yaml file: ", err)
+        return false, err
+    }
+    
+    _, err = exec.Command("chmod", "-R", "go+r", "./").CombinedOutput()
+    if err != nil {
+        logger.Error("unable to execute chmod: ", err)
+        return false, err
+    }
+    
+    logger.Debugf("Running Scorecard check for ",bundleImage)
+    
+    stdouterr, scorecarderr := exec.Command("operator-sdk", "scorecard",   
+                                  "--config", destinationFile,
+                                  "--selector=test=basic-check-spec-test",
+                                  "--output", "json", "bundle").CombinedOutput()
+    
+    scorecardFile := filepath.Join(artifactsDir,"/",scorecardBasicCheckResult)
+    
+    err = ioutil.WriteFile(scorecardFile, stdouterr, 0644)
+    if err != nil {
+        logger.Error("unable to copy result to /artifacts subdir: ", err)
+        return false, err
+    }
+
+    // we must send gojq a interface{}, so we have to convert our inspect output to that type
+    var inspectData interface{}
+    
+    err = json.Unmarshal(stdouterr, &inspectData)
+    if err != nil {
+        logger.Error("unable to parse scorecard json output")
+        logger.Debug("error Unmarshaling scorecard json output: ", err)
+        logger.Trace("failure in attempt to convert the raw bytes from `operator-sdk scorecard` to a interface{}")
+        return false, err
+    }
+
+    query, err := gojq.Parse(".items[].status.results[] | .name, .state")
+    if err != nil {
+        logger.Error("unable to parse scorecard json output")
+        logger.Debug("unable to parse :", err)
+        return false, err
+    }
+
+    // gojq expects us to iterate in the event that our query returned multiple matching values, but we only expect one.
+    iter := query.Run(inspectData)
+    
+    foundTestFailed := false
+
+    logger.Debug("scorecard output")
+    
+    for {
```

Comment:
To have a clean code, I suggest you have `return` statements inside the for loop, if it makes sense. Sth like:

 ```
for {
        v, ok := iter.Next()
        if !ok {
            break
        }
        if err, ok := v.(error); ok {
            logger.Error(err)
        }
        var s string = v.(string)
        logger.Debug(s)
        if strings.Contains(s, "fail") {
            return false, nil
      }
    }
return true, nil
```

Reply:
I can but this is a design intention to capture the tests in the output.
We can talk about this if you want to. I welcome suggestions of course.

Reply:
Of course! Let's book sometimes tomorrow, Friday July 2nd to talk about rewriting this block.